docs/content/Examples-Tutorials-Recipes/Entity-Attribute-Value.md

```diff
@@ -0,0 +1,334 @@
+  {
+    "UsersStatuses_Joins.name": "Ally Blanda",
+    "UsersStatuses_Joins.completedCreatedAt": "2019-03-05T00:00:00.000",
+    "UsersStatuses_Joins.processingCreatedAt": null,
+    "UsersStatuses_Joins.shippedCreatedAt": "2019-04-06T00:00:00.000"
+  },
+  {
+    "UsersStatuses_Joins.name": "Cayla Mayert",
+    "UsersStatuses_Joins.completedCreatedAt": "2019-06-14T00:00:00.000",
+    "UsersStatuses_Joins.processingCreatedAt": "2021-05-20T00:00:00.000",
+    "UsersStatuses_Joins.shippedCreatedAt": null
+  },
+  {
+    "UsersStatuses_Joins.name": "Concepcion Maggio",
+    "UsersStatuses_Joins.completedCreatedAt": null,
+    "UsersStatuses_Joins.processingCreatedAt": "2020-07-14T00:00:00.000",
+    "UsersStatuses_Joins.shippedCreatedAt": "2019-07-19T00:00:00.000"
+  }
+]
+```
+
+The drawback is that when the set of statuses changes, we'll need to amend the cube definition in several places: update selected values and joins in SQL as well as update the dimensions. Let's see how to work around that.
+
+### Static attributes, DRY version
+
+We can embrace the
+[Don't repeat yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle
+and eliminate the repetition by generating the cube definition dynamically based on the
+list of statuses. Let's move all repeated code patterns into handy functions and iterate
+over statuses in relevant parts of the cube's code.
+
+```javascript
+const statuses = [
+  'completed',
+  'processing',
+  'shipped'
+];
+
+const createValue = (status, index) =>
+  `MIN(orders_${index}.created_at) AS createdAt_${index}`
+
+const createJoin = (status, index) =>
+  `LEFT JOIN public.orders AS orders_${index}
+    ON users.id = orders_${index}.user_id
+    AND orders_${index}.status = '${status}'`
+
+const createDimension = (status, index) => ({
+  [`${status}CreatedAt`]: {
+    sql: (CUBE) => `createdAt_${index}`,
+    type: `time`,
+  }
+})
+
+cube(`UsersStatuses_DRY`, {
+  sql: `
+    SELECT
+      users.first_name,
+      users.last_name,
+      ${statuses.map(createValue).join(',')}
+    FROM public.users AS users
+    ${statuses.map(createJoin).join('')}
+    GROUP BY 1, 2
+  `,
+  
+  dimensions: Object.assign(
+    {
+      name: {
+        sql: `first_name || ' ' || last_name`,
+        type: `string`
+      }
+    },
+    statuses.reduce((all, status, index) => ({
+      ...all,
+      ...createDimension(status, index)
+    }), {})
+  )
+});
+```
+
+The new `UsersStatuses_DRY` cube is functionally identical to the `UsersStatuses_Joins`
+cube above. So, querying this new cube would yield the same data. However, there's still
+a static list of statuses present in the cube's cube. Let's see how to work around that.
+
+### Dynamic attributes
+
+We can eliminate the list of statuses from the cube's code by loading this list from an
+external source. Let's assume that we've created an API service that fetches statuses,
+obviously, by running a Cube query. Here's the code from the `fetch.js` file that defines
```

Comment:
I think this sentence would sound better without `, obviously,` as it's not necessary (or it may even sound a bit condescending to some).